src/System.Runtime/tests/System/GCTests.netcoreapp.cs

```diff
@@ -23,5 +24,71 @@ public static void GetAllocatedBytesForCurrentThread(int size)
             Assert.True((end - start) > size, $"Allocated too little: start: {start} end: {end} size: {size}");
             Assert.True((end - start) < 5 * size, $"Allocated too much: start: {start} end: {end} size: {size}");
         }
+
+        [Fact]
+        public static void GetGCMemoryInfo()
+        {
+            GCMemoryInfo memoryInfo = GC.GetGCMemoryInfo();
+
+            Assert.True(memoryInfo.HighMemoryLoadThresholdBytes > 0);
+            Assert.True(memoryInfo.MemoryLoadBytes > 0);
+            Assert.True(memoryInfo.TotalAvailableMemoryBytes > 0);
+            Assert.True(memoryInfo.HeapSizeBytes > 0);
+            Assert.True(memoryInfo.FragmentedBytes >= 0);
+        }
+
+        [Fact]
+        public static void GetGCMemoryInfoFragmentation()
+        {
+            GCHandle[] gch = null;
+
+            try
+            {
+                gch = new GCHandle[100];
+
+                for (int i = 0; i < gch.Length * 2; ++i)
+                {
+                    byte[] arr = new byte[128];
+                    if (i % 2 == 0)
+                    {
+                        gch[i / 2] = GCHandle.Alloc(arr, GCHandleType.Pinned);
+                    }
+                }
+
+                GC.Collect();
```

Comment:
It is possible that this `GC.Collect` will fill the holes in between the pinned arrays with some other allocations, and thus eliminating the fragmentation. I think this test needs to keeps trying to introduce the fragmentation to be robust.

Also, it may be a good idea for these tests to run in a separate process (look for `RemoteExecutor` on how to do this) to avoid interactions with other tests in running in parallel.